Makefile

```diff
@@ -1,92 +1,236 @@
+GLIDE_COMMIT := 84607742b10f492430762d038e954236bbaf23f7
+BTCD_COMMIT := $(shell cat Gopkg.toml | \
+		grep -A1 $(BTCD_PKG) | \
+		tail -n1 | \
+		awk '{ print $$3 }' | \
+		tr -d '"')
+
+GOBUILD := go build -v
+GOINSTALL := go install -v
+GOTEST := go test -v
+
+GOLIST := go list $(PKG)/... | grep -v '/vendor/'
+GOLISTCOVER := $(shell go list -f '{{.ImportPath}}' ./... | sed -e 's/^$(ESCPKG)/./')
+GOLISTLINT := $(shell go list -f '{{.Dir}}' ./... | grep -v 'lnrpc')
+
+COVER = for dir in $(GOLISTCOVER); do \
+		$(GOTEST) $(TEST_FLAGS) \
+			-covermode=count \
+			-coverprofile=$$dir/profile.tmp $$dir; \
+		\
+		if [ -f $$dir/profile.tmp ]; then \
+			cat $$dir/profile.tmp | \
+				tail -n +2 >> profile.cov; \
+			$(RM) $$dir/profile.tmp; \
+		fi \
+	done
+
+LINT = $(LINT_BIN) \
+	--disable-all \
+	--enable=gofmt \
+	--enable=vet \
+	--enable=golint \
+	--line-length=72 \
+	--deadline=4m $(GOLISTLINT) 2>&1 | \
+	grep -v 'ALL_CAPS\|OP_' 2>&1 | \
+	tee /dev/stderr
+
+CGO_STATUS_QUO := ${CGO_ENABLED}
+
+RM := rm -f
+CP := cp
+MAKE := make
+XARGS := xargs -L 1
+
+GREEN := "\\033[0;32m"
+NC := "\\033[0m"
+define print
+	echo $(GREEN)$1$(NC)
+endef
+
+include make/testing_flags.mk
+
+default: scratch
+
+all: scratch check install
+
+# ============
+# DEPENDENCIES
+# ============
+
+$(DEP_BIN):
+	@$(call print, "Fetching dep.")
+	go get -u $(DEP_PKG)
+
+$(GLIDE_BIN):
+	@$(call print, "Fetching glide.")
+	go get -d $(GLIDE_PKG)
+	cd ${GOPATH}/src/$(GLIDE_PKG) && git checkout $(GLIDE_COMMIT) 
+	$(GOINSTALL) $(GLIDE_PKG)
+
+$(GOVERALLS_BIN):
+	@$(call print, "Fetching goveralls.")
+	go get -u $(GOVERALLS_PKG)
+
+$(LINT_BIN):
+	@$(call print, "Fetching gometalinter.v1")
+	go get -u $(LINT_PKG)
+	$(GOINSTALL) $(LINT_PKG)
+
+dep: $(DEP_BIN)
+	@$(call print, "Compiling dependencies.")
+	dep ensure -v
+
+$(BTCD_DIR):
+	@$(call print, "Fetching btcd.")
+	go get -d github.com/roasbeef/btcd
 
+btcd: $(GLIDE_BIN) $(BTCD_DIR)
+	@$(call print, "Compiling btcd dependencies.")
+	cd $(BTCD_DIR) && git checkout $(BTCD_COMMIT) && glide install
+	@$(call print, "Installing btcd and btcctl.")
+	$(GOINSTALL) $(BTCD_PKG)
+	$(GOINSTALL) $(BTCD_PKG)/cmd/btcctl
 
-# ==============================================================================
+# ============
 # INSTALLATION
-# ==============================================================================
-
-deps:
-ifndef HAVE_DEP
-	@echo "Fetching dep."
-	go get -u github.com/golang/dep/cmd/dep
-endif
-	@echo "Building dependencies."
-	dep ensure -v
+# ============
 
-build: 
-	@echo "Building lnd and lncli."
-	go build -v -o lnd $(LDFLAGS) $(PKG)
-	go build -v -o lncli $(LDFLAGS) $(PKG)/cmd/lncli
+build:
+	@$(call print, "Building lnd and lncli.")
+	$(GOBUILD) -o lnd $(LDFLAGS) $(PKG)
+	$(GOBUILD) -o lncli $(LDFLAGS) $(PKG)/cmd/lncli
 
 install:
-	@echo "Installing lnd and lncli."
-	go install -v $(LDFLAGS) $(PKG)
-	go install -v $(LDFLAGS) $(PKG)/cmd/lncli
+	@$(call print, "Installing lnd and lncli.")
+	$(CP) ./lnd $(LND_BIN)
```

Comment:
`LND_BIN` and `LNCLI_BIN` aren't defined. IMO this should go back to just using `go install` like before, this'll re-use the binary in place if its already been built.

Reply:
Fixed 👍